iidm/iidm-api/src/main/java/com/powsybl/iidm/network/TapChanger.java

```diff
@@ -6,16 +6,13 @@
  */
 package com.powsybl.iidm.network;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
-import java.util.OptionalInt;
+import java.util.*;
 
 /**
  *
  * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
  */
-public interface TapChanger<C extends TapChanger<C, S>, S extends TapChangerStep<S>> {
+public interface TapChanger<C extends TapChanger<C, S, R>, S extends TapChangerStep<S>, R extends TapChangerStepsReplacer<R, ?>> {
```

Comment:
You can remove the wildcard by using another generic type here. 
Something like :
public interface TapChanger<C extends TapChanger<C, S, A, R>, S extends TapChangerStep<S>, A extends TapChangerStepAdder<A, R>, R extends TapChangerStepsReplacer<R, A>> {

Reply:
Done